### PR TITLE
Go

### PR DESCRIPTION
**Pull request by cewood**

### CODE REVIEW

File: go/supermarket/shopping_cart.go [hunk 1]
```diff
@@ -36,46 +33,42 @@ func (c *ShoppingCart) addItemQuantity(product Product, amount float64) {
 	}
 }
 
+// TODO: move this to the teller, it's similar to checking out
```
Comment (emilybache) [inline go/supermarket/shopping_cart.go:36]: Don't give people clues!

File: go/supermarket/shopping_cart.go [hunk 2]
```diff
@@ -36,46 +33,42 @@ func (c *ShoppingCart) addItemQuantity(product Product, amount float64) {
...
+		if offer, ok := offers[p]; ok {
 
-			} else if offer.offerType == TwoForAmount {
-				x = 2
-				if quantityAsInt >= 2 {
-					var total = offer.argument * float64(quantityAsInt / x) + float64(quantityAsInt % 2) * unitPrice
-					var discountN = unitPrice * quantity - total;
-					discount = &Discount{product: p, description: fmt.Sprintf("2 for %.2f", offer.argument), discountAmount: -discountN}
-				}
+			// check all the products in an offer, in case it's a bundle
```
Comment (emilybache) [inline go/supermarket/shopping_cart.go:44]: honestly! this is supposed to be legacy code!

**Comment by emilybache** [comment]
Sorry, but this is a solution not an upgrade of the go version. I won't accept it since I want people to be able to do the exercise themselves!

**Comment by cewood** [comment]
> Sorry, but this is a solution not an upgrade of the go version. I won't accept it since I want people to be able to do the exercise themselves!

Apologies, mistakenly opened this PR against the wrong repo/upstream :see_no_evil:

**Pull request closed without merging**